scaladoc/src/dotty/tools/scaladoc/tasty/SymOps.scala

```diff
@@ -119,21 +152,27 @@ class SymOps[Q <: Quotes](val q: Q):
           else if (sym.maybeOwner.isDefDef) Some(sym.owner)
           else None
 
-        val originPath = {
+        val className = sym.className
+
+        val location = className.fold(sym.packageName)(cn => s"${sym.packageName}.${cn}")
+
+        val anchor = sym.anchor
+
+        val externalLink = {
             import q.reflect._
             import dotty.tools.dotc
             given ctx: dotc.core.Contexts.Context = q.asInstanceOf[scala.quoted.runtime.impl.QuotesImpl].ctx
             val csym = sym.asInstanceOf[dotc.core.Symbols.Symbol]
-            Option(csym.associatedFile).fold("")(_.path)
+            Option(csym.associatedFile).map(_.path).flatMap( path =>
```

Comment:
Did you notice any slowdown after these changes? especially this line may affect performance so we may consider caching (this needs to be tested with scala 3 doc not, the `self` one)